### PR TITLE
fix(fe-writer) Add missing JS files(changes_loader and changes_preview ) as ts files

### PR DESCRIPTION
**Pull request by devinea**
fix for #676

Because of the ignore here 
https://github.com/SAP/open-ux-tools/blob/main/packages/fiori-elements-writer/src/index.ts#L102
the js files are not copied. Adding a TS copy of the files so that the copy works. 
transpiling middleware removes the comments but otherwise seems ok. 
Added a test to cover the case

### CODE REVIEW

**Review by IainSAP — APPROVED** [review]
Thx @devinea

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
Please fix the type errors (`Found 46 errors in 3 files.`). To see them just run in `packages/fiori-elements-writer`
```
UX_DEBUG=x pnpm test
cd test/test-output/lropTemplates/lropV2_ts
npm i
npm run ts-typecheck
```

File: packages/fiori-elements-writer/test/lrop.test.ts [hunk 1]
```diff
@@ -135,6 +135,26 @@ describe(`Fiori Elements template: ${TEST_NAME}`, () => {
                 service: v2Service
             } as FioriElementsApp<LROPSettings>
         }
+        ,
+        {
+            name: 'lropV2_ts',
+            config: {
+                ...Object.assign(feBaseConfig('felrop4'), {
```
Comment (tobiasqueck) [inline packages/fiori-elements-writer/test/lrop.test.ts:142]: Can you change this one to 'lropV2_ts' as well. (not critical but confused me when looking at the generated project).

File: packages/fiori-elements-writer/test/lrop.test.ts [hunk 2]
```diff
@@ -135,6 +135,26 @@ describe(`Fiori Elements template: ${TEST_NAME}`, () => {
...
+                    template: {
+                        type: TemplateType.ListReportObjectPage,
+                        settings: v2TemplateSettings
+                    },
+                    ui5: {
+                        ...feBaseConfig('felrop4'),
```
Comment (tobiasqueck) [inline packages/fiori-elements-writer/test/lrop.test.ts:148]: Line 148 is not clear to me.

File: packages/fiori-elements-writer/test/lrop.test.ts (hunk 1 above)
Comment (devinea) [reply packages/fiori-elements-writer/test/lrop.test.ts]: Done

File: packages/fiori-elements-writer/test/lrop.test.ts (hunk 2 above)
Comment (devinea) [reply packages/fiori-elements-writer/test/lrop.test.ts]: Removed

**Comment by devinea** [comment]
> Please fix the type errors (`Found 46 errors in 3 files.`). To see them just run in `packages/fiori-elements-writer`
> 
> ```
> UX_DEBUG=x pnpm test
> cd test/test-output/lropTemplates/lropV2_ts
> npm i
> npm run ts-typecheck
> ```

Screenshot 2022-09-14 at 20 54 34

It shows one error. I think you mentioned this is missing in the types module so we need an updated version with that added?
OK to merge in the meantime or we should add ts ignore / ts no check in that case for now?

**Comment by tobiasqueck** [comment]
> It shows one error. I think you mentioned this is missing in the types module so we need an updated version with that added?
> OK to merge in the meantime or we should add ts ignore / ts no check in that case for now?

Yes, I have reached out to the corresponding team already and hope it get fixed soon.

**Review by tobiasqueck — APPROVED** [review]
I have to admit, I expected a different fix than `// @ts-nocheck` but since these files are rather static, I am ok with it.

**Review by IainSAP — APPROVED** [review]
Thx @devinea 

- New files added as expected
- Tests to cover

**Comment by devinea** [comment]
> I have to admit, I expected a different fix than `// @ts-nocheck` but since these files are rather static, I am ok with it.

I looked into adding `exclude` to the `tsconfig` for `webapp/test` like we discussed and this worked ok for `npm run ts-typecheck`. But in VS Code it showed various TS errors in these files. `// @ts-nocheck`  gives a consistent result.

I also looked into converting these files to full TS, but there was various errors with types not exposed so seem safer to ignore them for now. I think these will be removed eventually when middleware takes care of this for V2 projects like in V4.

**Pull request merged**